assets/js/googlesitekit/datastore/site/first-party-mode.js

```diff
@@ -0,0 +1,184 @@
+ */
+import API from 'googlesitekit-api';
+import {
+	commonActions,
+	createRegistrySelector,
+	combineStores,
+} from 'googlesitekit-data';
+import { CORE_SITE } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+
+const SET_FIRST_PARTY_MODE_ENABLED = 'SET_FIRST_PARTY_MODE_ENABLED';
+
+const fetchGetFirstPartyModeSettings = createFetchStore( {
+	baseName: 'getFirstPartyModeSettings',
+	controlCallback: () =>
+		API.get( 'core', 'site', 'fpm-settings', undefined, {
+			useCache: false,
+		} ),
+	reducerCallback: ( state, firstPartyModeSettings ) => {
+		return {
+			...state,
+			firstPartyModeSettings: {
+				...( state.firstPartyModeSettings || {} ),
+				...firstPartyModeSettings,
+			},
+		};
+	},
+} );
+
+const fetchSaveFirstPartyModeSettings = createFetchStore( {
+	baseName: 'saveFirstPartyModeSettings',
+	controlCallback: ( { settings } ) => {
+		return API.set( 'core', 'site', 'fpm-settings', { settings } );
+	},
+	reducerCallback: ( state, firstPartyModeSettings ) => {
+		return {
+			...state,
+			firstPartyModeSettings: {
+				...( state.firstPartyModeSettings || {} ),
+				...firstPartyModeSettings,
+			},
+		};
+	},
+	argsToParams: ( settings ) => {
+		return { settings };
+	},
+	validateParams: ( { settings } ) => {
+		invariant(
+			isPlainObject( settings ),
+			'settings must be a plain object.'
+		);
+	},
+} );
+
+const baseInitialState = {
+	firstPartyModeSettings: undefined,
+};
+
+const baseActions = {
+	setFirstPartyModeEnabled() {
+		return {
+			type: SET_FIRST_PARTY_MODE_ENABLED,
+		};
+	},
+};
+
+const baseControls = {};
+
+const baseReducer = ( state, { type } ) => {
+	switch ( type ) {
+		case SET_FIRST_PARTY_MODE_ENABLED: {
+			return {
+				...state,
+				firstPartyModeSettings: {
+					...( state.firstPartyModeSettings || {} ),
+					enabled: true,
+				},
+			};
+		}
+
+		default: {
+			return state;
+		}
+	}
+};
+
+const baseResolvers = {
+	*getFirstPartyModeSettings() {
+		const { select } = yield commonActions.getRegistry();
+
+		const settings = select( CORE_SITE ).getFirstPartyModeSettings();
+		if ( settings === undefined ) {
+			yield fetchGetFirstPartyModeSettings.actions.fetchGetFirstPartyModeSettings();
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the first-party mode settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} First-party mode settings, or `undefined` if not loaded.
+	 */
+	getFirstPartyModeSettings: ( state ) => {
+		return state.firstPartyModeSettings;
+	},
+
+	/**
+	 * Checks if first-party mode is enabled.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} True if first-party mode is enabled, otherwise false.
+	 */
+	isFirstPartyModeEnabled: createRegistrySelector( ( select ) => () => {
+		const { firstPartyModeSettings } =
+			select( CORE_SITE ).getFirstPartyModeSettings();
+		return !! firstPartyModeSettings?.enabled;
```

Comment:
We should return `undefined` if the `firstPartyModeSettings` is not defined to indicate the state is still loading... The same applies to the selectors below.